packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -982,3 +981,142 @@ function getNativeQemuArch() {
+        "system",
+        "system_locales",
+        locale,
+      ]);
+
+      // this is needed to make sure that changes will persist
+      await exec(ADB_PATH, ["-s", this.serial!, "shell", "sync"]);
+
+      // TODO:  Find a way to change or remove persist.sys.locale without root access. note: removing the whole
+      // data/property/persistent_properties file would also work as we persist device settings globally in Radon IDE
+      const { stdout } = await exec(ADB_PATH, [
+        "-s",
+        this.serial!,
+        "shell",
+        "getprop",
+        "persist.sys.locale",
+      ]);
+
+      if (stdout) {
+        Logger.warn(
+          "Updating locale will not take effect as the device has altered locale via system settings which always takes precedence over the device setting the IDE uses."
+        );
+      }
+
+      // locale changes require boot to take effect
+      return { bootChanged: true };
+    } catch (e) {
+      Logger.error("Failed to update locale settings", e);
+      return { bootChanged: false };
+    }
+  }
+}
+
+class CameraSettingApplier implements SettingApplier {
+  constructor(private cameraSettings: DeviceSettings["camera"]) {}
+
+  async shouldUpdate(configLines: string[]): Promise<boolean> {
+    try {
+      let currentBackCamera = "emulated";
+      let currentFrontCamera = "none";
+
+      configLines.forEach((line: string) => {
+        const [key, value] = line.split("=");
+        if (key === "hw.camera.back") {
+          currentBackCamera = value;
+        } else if (key === "hw.camera.front") {
+          currentFrontCamera = value;
+        }
+      });
+
+      const backNeedsUpdate =
+        this.cameraSettings?.back !== undefined && currentBackCamera !== this.cameraSettings.back;
+      const frontNeedsUpdate =
+        this.cameraSettings?.front !== undefined &&
+        currentFrontCamera !== this.cameraSettings.front;
+
+      return backNeedsUpdate || frontNeedsUpdate;
+    } catch (e) {
+      Logger.warn("Failed to read AVD config for camera settings comparison", e);
+      return false;
+    }
+  }
+
+  async updateSettings(
+    configLines: string[]
+  ): Promise<{ bootChanged: boolean; updatedConfigLines: string[] }> {
+    try {
+      let hasChanges = false;
+      for (let i = 0; i < configLines.length; i++) {
+        const [key] = configLines[i].split("=");
+        if (key === "hw.camera.back" && this.cameraSettings?.back !== undefined) {
+          configLines[i] = `hw.camera.back=${this.cameraSettings.back}`;
+          hasChanges = true;
+        } else if (key === "hw.camera.front" && this.cameraSettings?.front !== undefined) {
+          configLines[i] = `hw.camera.front=${this.cameraSettings.front}`;
+          hasChanges = true;
+        }
+      }
+
+      // camera changes require boot to take effect
+      return { bootChanged: hasChanges, updatedConfigLines: configLines };
+    } catch (e) {
+      Logger.error("Failed to update AVD camera settings", e);
```

Comment:
when can this even throw? We seem to be just doing some string concatenation & dict updates. Also, if any of that really throws, I don't think we should still be updating the config lines, this could mean that we'd updated the config partially which may not be ideal